include/s3select_parquet_intrf.h

```diff
@@ -747,6 +747,15 @@ class PARQUET_EXPORT RowGroupReader {
   std::unique_ptr<Contents> contents_;
 };
 
+//TODO external setting? RGW options ??
+#define RGW_buffer_size 1024*1024*16
```

Comment:
can you pass it as a parameter to the function that init the parquet reader?
should probably be an RGW conf parameter passed from the outside

Reply:
@yuvalif 
sure. 
that is the plan. (review the comment above)

i measured whether it resolved the issue of huge memory consumption.

Reply:
ok. but we should probably merge after we make the change to take the size as a parameter, so we can integrate into the rgw code

Reply:
yes, the interface (RGW option) is included in this PR.
 
with the standalone-application the impact of this change is visible(memory consumption, RSS).
it is also possible to measure other aspects of these changes, like the number of calls to storage systems.  

this parameter probably has more impact upon using the S3 storage, we should strive for a default size that is optimized with RSS and throughput. 
(users should not "play" with this parameter too much, this may cause other issues.)




with this

Reply:
how can an RGW option be included in an s3-select PR?
I thought that this PR would expose an interface where this value could be set, and the matching RGW PR (https://github.com/ceph/ceph/pull/59465) will add a new option in: `src/common/options/rgw.yaml.in` and pass it to that API